server/src/service.rs

```diff
@@ -293,4 +314,42 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &[OwnedObjectPath],
+    ) -> Result<Vec<OwnedObjectPath>, ServiceError> {
+        let mut locked_changed = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for object in objects {
+            for collection in collections.iter() {
+                if object == collection.path() {
+                    if locked == collection.is_locked().await {
```

Comment:
this is wrong.... If you want to lock an already locked collection, then there is nothing to change and it should be skipped. So the comment below is ok, but not the `locked_changed.push()` part. Same for the Item.

Reply:
No. The `locked_changed` is there to track/filter-out objects locked/unlocked without a prompt.